src/autorest.bicep/src/sample-generator.ts

```diff
@@ -104,6 +105,10 @@ function getBicep(value: any, indent: number) {
   throw new Error(`Unsupported type: ${typeof value}`);
 }
 
+function isBicepIdentifier(value: string) {
```

Comment:
Is this a reliable way to tell the difference between the two cases (e.g. `basicString` vs `special.char-property`)? I thought there would be a difference between the definitions between the two in API spec.

Reply:
I could be misunderstanding, but the goal with this PR is just to ensure we're generating valid Bicep code. The Bicep lexer use similar logic when tokenizing, so here I just want to ensure that if a particular key needs escaping in Bicep, then it's emitted as a string.

Here's what Bicep's lexing logic looks like for handling identifiers: https://github.com/Azure/bicep/blob/95ede42853822fb0a7c88380061a4b4590fa57cd/src/Bicep.Core/Parsing/Lexer.cs#L838-L851

When lexing, here's how Bicep would interpret various examples:
`"allOneProperty" -> identifier(allOneProperty)`
`"abc.def" -> identifier(abc) dot(.) identifier(def)`

When parsing, Bicep expects the object key to either be a single identifier token or a single string token - the grammar looks like this:
`(identifier | string) ":" expression`